src/node_http2.cc

```diff
@@ -778,6 +805,27 @@ inline int32_t GetFrameID(const nghttp2_frame* frame) {
       frame->hd.stream_id;
 }
 
+inline int Http2Session::OnBeforeFrameSend(nghttp2_session* handle,
+                                           const nghttp2_frame* frame,
+                                           void* user_data) {
+  Http2Session* session = static_cast<Http2Session*>(user_data);
+  int32_t id = GetFrameID(frame);
+  Http2Stream* stream = session->FindStream(id);
+  switch (frame->hd.type) {
+    case NGHTTP2_HEADERS:
+      TRACE_EVENT_NESTABLE_ASYNC_BEGIN1(
```

Comment:
are there cases where this `Send Headers` Begin event could not be paired with a corresponding end event (e.g., if `OnFrameNotSent` is called)?.  What are implications of if only a begin event happens w/out corresponding end event? (and vice versa)?

Reply:
*Potentially* yes, and will need to tease those conditions out. It's also not clear *when* this particular span should begin. There are two separate moments that are important: 1) the moment the user enqueues the headers to be sent and 2) the moment nghttp2 actually begins sending those headers. It's not clear which is going to be the most useful for users. For myself, as the person writing the core http2 code, knowing exactly when nghttp2 begins sending the headers helps with tuning core performance, but it's likely not particularly helpful for end users.